src/ips_options/ips_sd_pattern.cc

```diff
@@ -303,7 +302,11 @@ static const Parameter s_params[] =
 class SdPatternModule : public Module
 {
 public:
-    SdPatternModule() : Module(s_name, s_help, s_params) { }
+    SdPatternModule() : Module(s_name, s_help, s_params)
+    {
+        scratch_index = SnortConfig::request_scratch(SdPatternModule::scratch_setup,
+                            SdPatternModule::scratch_cleanup);
```

Comment:
Nit - 4 space indent; one possible way to go:

        scratch_index = SnortConfig::request_scratch(
            SdPatternModule::scratch_setup, SdPatternModule::scratch_cleanup);